app/models/friendship.rb

```diff
@@ -1,18 +1,20 @@
 class Friendship < ApplicationRecord
-  after_create :create_mutual_relationship
-  after_destroy :destroy_mutual_relationship
+  after_create :delete_request
 
-  belongs_to :user
-  belongs_to :friend, class_name: 'User'
+  belongs_to :sent_friend, class_name: 'User', foreign_key: 'adder_id'
+  belongs_to :received_friend, class_name: 'User', foreign_key: 'added_id'
+
+  validate :friend_is_self
 
   private
 
-  def create_mutual_relationship
-    Friendship.find_or_create_by(user_id: friend.id, friend_id: user.id)
+  def friend_is_self
```

Comment:
There's already a method in `friend_request.rb` that safeguards against a user sending a friend request to him/herself. Probably with that method in place, this one will never be run. In that case we can safely delete it perhaps?